keps/sig-scheduling/4328-node-affinity-based-eviction/README.md

```diff
@@ -0,0 +1,479 @@
+- Test that `RequiredDuringSchedulingRequiredDuringExecution` node affinity is ignored during scheduling when the feature is disabled.
+- Test that DaemonSet with `RequiredDuringSchedulingRequiredDuringExecution` does not enter a scheduling and descheduling hot loop after being evicted. 
+
+##### e2e tests
+
+These tests will be added:
+- `RequiredDuringSchedulingRequiredDuringExecution` will be tested during both scheduling and execution.
+  1. create a node that has a label
+  2. create a pod with `RequiredDuringSchedulingRequiredDuringExecution` node affinity matches the node.
+  3. verify that the pod is correctly scheduled to the node
+  4. remove the node label
+  5. verify that the pod is evicted
+  6. create a new pod with same template
+  7. verify that the pod failed scheduling
+  8. create a new node which label matches the pod's node-affinity
+  9. verify that the pod created in 6 is scheduled to the new node
+   
+### Graduation Criteria
+
+#### Alpha
+
+- Feature implemented behind a feature flag
+- Initial unit/e2e tests completed and enabled
+- Documentation is added to demonstrate why this is useful how exactly affinity needs to be configured
+- Documentation is added to demonstrate potential pitfalls of the eviction and handling of these.
+- Initial metrics are added
+
+#### Beta
+
+- Gather feedback from developers and surveys
+- Additional e2e tests are completed(if needed)
+- Additional metrics are added(if needed) 
+- See if this kep can align with `Evacuation API`
+
+#### GA
+
+- No negative feedback
+- No bug issues reported
+
+### Upgrade / Downgrade Strategy
+
+#### Upgrade
+- Enable the feature gate in kube-apiserver, kube-controller-manager and kube-scheduler, and set `.spec.affinity.nodeAffinity.requiredDuringSchedulingRequiredDuringExecution` in pod.
+
+#### Downgrade
+
+- Disable the feature gate in kube-apiserver, kube-controller-manager and kube-scheduler, so that previously configured
+  `.spec.affinity.nodeAffinity.requiredDuringSchedulingRequiredDuringExecution` value will be ignored.
+- However, the `.spec.affinity.nodeAffinity.requiredDuringSchedulingRequiredDuringExecution` value of a Pod is preserved if it's previously configured; otherwise get silently dropped.
+
+### Version Skew Strategy
+
+kube-controller-manager, kube-scheduler and kube-apiserver will need to enable the feature gate for the full featureset
+to be present. 
+
+If kube-apiserver doesn't enable the feature gate, this new affinity will be silently dropped during pod creating, and will be silently dropped during pod updating if it's not been used.
+
+If kube-apiserver enable the feature gate, and:
+if only the kube-controller-manager enables the feature gate, the node affinity will be ignored during the scheduling phase.
+if only the kube-scheduler enables the feature gate, the node affinity will be ignored during the execution phase. 
```

Comment:
+ reverse here

Reply:
Not sure what you mean here😢.

Reply:
I think Filip is asking that you explain that the daemonset controller would create pods for it, even though they will be unschedulable.